src/lightning/fabric/strategies/fsdp.py

```diff
@@ -599,8 +599,18 @@ def _setup_activation_checkpointing(module: "FullyShardedDataParallel", layers:
         apply_activation_checkpointing,
         checkpoint_wrapper,
         CheckpointImpl,
+        CheckpointWrapper,
     )
 
+    if any(isinstance(mod, CheckpointWrapper) for mod in module.modules()):
+        if layers:
+            rank_zero_warn(
+                f"FSDP checkpointing for the layers {layers} is configured, but the model already contains checkpointed"
```

Comment:
We could probably improve the formatting of the message a bit here to avoid printing the raw class representation. 
```suggestion
                f"FSDP checkpointing for the layers {[layer.__name__ for layer in layers]} is configured, but the model already contains checkpointed"
```